cmd/clusterctl/client/cluster/proxy.go

```diff
@@ -70,9 +70,18 @@ func (k *proxy) NewClient() (client.Client, error) {
 		return nil, err
 	}
 
-	c, err := client.New(config, client.Options{Scheme: Scheme})
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to connect to the management cluster")
+	var c client.Client
+	// Nb. The operation is wrapped in a retry loop to make newClientSet more resilient to temporary connection problems.
+	connectBackoff := newConnectBackoff()
+	if err := retryWithExponentialBackoff(connectBackoff, func() error {
+		var err2 error
+		c, err2 = client.New(config, client.Options{Scheme: Scheme})
+		if err2 != nil {
+			return errors.Wrap(err2, "failed to connect to the management cluster")
```

Comment:
do we need to wrap here, or should we wrap on line 84 instead?

Reply:
done